pcsx2/Gif.cpp

```diff
@@ -85,9 +192,33 @@ __fi void gifInterrupt()
 		return;
 	}	
 
+	if (CHECK_GIFFIFOHACK) {
+		
+		if (int amtRead = gif_fifo.read(true)) {
+
+			if (!gifUnit.Path3Masked() || gifRegs.stat.FQC < 16) {
+				GifDMAInt(amtRead * BIAS);
+				return;
+			}
+		}
+		else {
+
+			if (!gifUnit.CanDoPath3() && gifRegs.stat.FQC == 16)
+			{
+				if (gifch.qwc > 0 || gspath3done == false) {
+					if (!gifUnit.Path3Masked()) {
```

Comment:
I could use that, but I had done this for readability so it was easy to see what was going on. I can change it if it's preferred.